service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -44,7 +51,11 @@ protected static boolean isLastPage(JSONObject response) throws JSONException {
         JSONArray dataArray = response.getJSONArray("data");
         for (int i = 0; i < dataArray.length(); i++) {
             JSONObject data = dataArray.getJSONObject(i);
-            results.put(data.getString("geoAreaCode"), new IndicatorValueFloat(data.getDouble("value")));
+            if(dataObjectValueTypeIsSupported(data)){
+                results.put(data.getString("geoAreaCode"), new IndicatorValueFloat(data.getDouble("value")));
```

Comment:
Since we already have GEO_AREA_CODE_KEY let's use it, `data.getString("geoAreaCode")` => `data.getString(GEO_AREA_CODE_KEY)`

Reply:
Updated code as suggested.